### PR TITLE
test: add C-API boundary test for empty strings

### PR DESCRIPTION
**Pull request by Sriram-B-Srivatsa**
### Description
This PR introduces a boundary case unit test for `WasmEdge_StringCreateByCString` within the `APICoreTest` suite. 

### Technical Details
While the C-API handles standard strings, boundary conditions like empty C-strings (`""`) must be explicitly verified to ensure:
1. The `WasmEdge_String.Length` is correctly initialized to `0`.
2. The internal buffer mapping remains consistent with `WasmEdge_StringWrap`.
3. Memory deallocation via `WasmEdge_StringDelete` handles these descriptors without regression.

This improves the reliability of the string-processing layer for host-function integrations.

### CODE REVIEW

File: test/api/APIUnitTest.cpp [hunk 1]
```diff
@@ -3136,6 +3136,17 @@ TEST(APICoreTest, Async) {
   WasmEdge_ExecutorDelete(Exec);
 }
 
+TEST(APICoreTest, StringEmptyTest) {
```
Comment (q82419) [inline test/api/APIUnitTest.cpp]: Please merge this test into `TEST(APICoreTest, String)`. Thanks.

Comment (Sriram-B-Srivatsa) [reply test/api/APIUnitTest.cpp]: Done! I've moved the empty string boundary checks inside the main TEST(APICoreTest, String) block and rebased against the latest master. Thank you for the review!

**Comment by Sriram-B-Srivatsa** [comment]
PR #4683: Noticed a Debian build failure with exit code 100. I think its an upstream apt-update network error.
